--- a/mediapipe/framework/api3/calculator_context.h
+++ b/mediapipe/framework/api3/calculator_context.h
@@ -431,8 +431,7 @@
   return std::forward<F>(visitor)(packet.Get<T>());
 }
 
-template <typename T, typename U, typename... Rest, int&... DoNotSpecify,
-          typename F>
+template <typename T, typename U, typename... Rest, typename F>
 auto VisitPacketOrDie(F&& visitor, const mediapipe::Packet& packet) {
   if (packet.ValidateAsType<T>().ok()) {
     return std::forward<F>(visitor)(packet.Get<T>());
@@ -447,8 +446,7 @@
   return std::forward<F>(visitor)(WrapLegacyPacket<T>(packet).value());
 }
 
-template <typename T, typename U, typename... Rest, int&... DoNotSpecify,
-          typename F>
+template <typename T, typename U, typename... Rest, typename F>
 auto VisitPacketAsPacketOrDie(F&& visitor, const mediapipe::Packet& packet) {
   if (packet.ValidateAsType<T>().ok()) {
     return std::forward<F>(visitor)(WrapLegacyPacket<T>(packet).value());
